src/cmd/support_archive/resource_query.go

```diff
@@ -0,0 +1,100 @@
+package support_archive
+
+import (
+	"reflect"
+
+	"github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/src/webhook"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type resourceQueryGroup struct {
+	resources []schema.GroupVersionKind
+	filters   []client.ListOption
+}
+
+type resourceQuery struct {
+	groupVersionKind schema.GroupVersionKind
+	filters          []client.ListOption
+}
+
+func getQueries(namespace string) []resourceQuery {
+	allQueries := make([]resourceQuery, 0)
+	allQueries = append(allQueries, getInjectedNamespaceQueryGroup().getQueries()...)
+	allQueries = append(allQueries, getOperatorNamespaceQueryGroup(namespace).getQueries()...)
+	allQueries = append(allQueries, getOperatorComponentsQueryGroup(namespace).getQueries()...)
+	return allQueries
+}
+
+func getInjectedNamespaceQueryGroup() resourceQueryGroup {
+	return resourceQueryGroup{
+		resources: []schema.GroupVersionKind{
+			toGroupVersionKind(corev1.SchemeGroupVersion, corev1.Namespace{}),
+		},
+		filters: []client.ListOption{
+			client.HasLabels{
+				webhook.InjectionInstanceLabel,
+			},
+		},
+	}
+}
+
+func getOperatorNamespaceQueryGroup(namespace string) resourceQueryGroup {
+	return resourceQueryGroup{
+		resources: []schema.GroupVersionKind{
+			toGroupVersionKind(corev1.SchemeGroupVersion, corev1.Namespace{}),
+		},
+		filters: []client.ListOption{
+			client.MatchingFields{
+				"metadata.name": namespace,
+			},
+		},
+	}
+}
+
+func getOperatorComponentsQueryGroup(namespace string) resourceQueryGroup {
+	return resourceQueryGroup{
+		resources: []schema.GroupVersionKind{
+			toGroupVersionKind(appsv1.SchemeGroupVersion, appsv1.Deployment{}),
+			toGroupVersionKind(appsv1.SchemeGroupVersion, appsv1.StatefulSet{}),
+			toGroupVersionKind(appsv1.SchemeGroupVersion, appsv1.DaemonSet{}),
+			toGroupVersionKind(appsv1.SchemeGroupVersion, appsv1.ReplicaSet{}),
+			toGroupVersionKind(corev1.SchemeGroupVersion, corev1.Service{}),
+			toGroupVersionKind(corev1.SchemeGroupVersion, corev1.Pod{}),
+			toGroupVersionKind(v1beta1.GroupVersion, v1beta1.DynaKube{}),
+		},
+		filters: []client.ListOption{
+			client.MatchingLabels{
+				kubeobjects.AppNameLabel: "dynatrace-operator",
+			},
+			client.InNamespace(namespace),
```

Comment:
so we collect only stuff that belongs directly to the operator?
I think it would be preferable to collect everything in the operator's namespace, as that will give us a fuller picture of what we are dealing with.

Reply:
that said, there still could be some 3rd party stuff in the operator's namespace, that we will not know anything about because it can be a not-standard component so we couldn't really added to our list even if we wanted to.

(!! be warned, bad idea incoming !!)
we could get all the crds in the cluster and list every cr in our namespace.
would this blow up the resource usage of the support archive collection? most definitely 
should we do this? most definitely not :)

Reply:
> we could get all the crds in the cluster and list every cr in our namespace.

would require the operator to have permission to do this (non-namespaced CRDs and namespaced CRs), which sounds like a bad idea, just for the fact that it _may_ be useful in the support archive.

Reply:
> (!! be warned, bad idea incoming !!)
I did say that it was a bad idea 😂

Reply:
TLDR: Its fine as it is now